### PR TITLE
chore(RHTAPWATCH-741): Add recording rules for exporter

### PR DESCRIPTION
**Pull request by hmariset**
Add recording rules to translate the components existing metrics to expose availability.

### CODE REVIEW

File: rhobs/recording/exporter_recording_rules.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+groups:
+- name: exporter
+  rules:
+  - record: konflux_up
+    expr: label_replace(grafana_ds_up, "service", "grafana_copy", "__name__", "(.+)")
```
Comment (amisstea) [inline rhobs/recording/exporter_recording_rules.yml:5]: Why refer to the service as `grafana_copy` instead of `grafana`?

**Review by amisstea — COMMENTED** [review]
Do you plan to add some documentation in a followup PR (building upon #173)?

File: rhobs/recording/exporter_recording_rules.yml (hunk 1 above)
Comment (hmariset) [reply rhobs/recording/exporter_recording_rules.yml]: @amisstea updated it.Please check.

**Review by amisstea — APPROVED** [review]
LGTM

File: .tekton/o11y-pull-request.yaml [hunk 2]
```diff
@@ -280,7 +280,7 @@ spec:
         - name: url
           value: https://github.com/tektoncd/catalog.git
         - name: revision
-          value: main
+          value: f4708d478ee8fac6b5b68347cde087cb7c1d1b1c
```
Comment (yftacherzog) [inline .tekton/o11y-pull-request.yaml:283]: What is this change about?

**Review by yftacherzog — CHANGES_REQUESTED** [review]
- We need test cases for the recording rules. You can find references for recording rules test cases in our git history (we used to have a couple of recording rules in the past and you should be able to see references to their test cases).
- I'm concerned for the fact that the metric name can change without us having to change the test cases (we might be missing test cases). Not sure if this PR is the place to address that, but please spend some time examining this and open a ticket to address/explore it.

File: .tekton/o11y-pull-request.yaml (hunk 2 above)
Comment (yftacherzog) [reply .tekton/o11y-pull-request.yaml:283]: I created [another PR](https://github.com/redhat-appstudio/o11y/pull/209) for this. You can remove the changes here and rebase.

Comment (hmariset) [reply .tekton/o11y-pull-request.yaml:283]: EC was failing with violation `msg: Task kube-linter is used by pipeline task run-kubelint via an unpinned reference` so I have updated it to use recent commit sha from the task used over there.

Sure @yftacherzog, I'll remove the changes here

File: test/promql/tests/recording/exporter_recording_rules_test.yml
Comment (yftacherzog) [inline test/promql/tests/recording/exporter_recording_rules_test.yml:1]: Are those tests being executed for the PR? Looking at the makefile, I suspect they do not.
Make sure you're able to make the PR checks fail with the recording rules tests.

File: test/promql/tests/recording/exporter_recording_rules_test.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+evaluation_interval: 1m
+
+rule_files:
+- ../../../../rhobs/recording/exporter_recording_rules.yml
+
+tests:
+- interval: 1m
+  name: DSExporterTest
+  input_series:
```
Comment (yftacherzog) [inline test/promql/tests/recording/exporter_recording_rules_test.yml:9]: Can you also add a test case in which the `grafana_ds_up` gives no output (e.g. only `_`s)?

Comment (hmariset) [reply test/promql/tests/recording/exporter_recording_rules_test.yml:9]: not relevant anymore

File: test/promql/tests/recording/exporter_recording_rules_test.yml
Comment (hmariset) [reply test/promql/tests/recording/exporter_recording_rules_test.yml:1]: Updated the makefile to execute recording rules

**Comment by hmariset** [comment]
@yftacherzog @amisstea I have updated the code. Please review again.

**Review by amisstea — APPROVED** [review]

**Review by yftacherzog — APPROVED** [review]
Looks good!
Once you have this one merged, make sure you create the necessary changes in app-interface and update the [readme here](https://github.com/redhat-appstudio/o11y?tab=readme-ov-file#updating-alerts) with the instructions for how to update the recording rules references.

**Review by avi-biton — APPROVED** [review]
/lgtm

**Comment by hmariset** [comment]
> Looks good! Once you have this one merged, make sure you create the necessary changes in app-interface and update the [readme here](https://github.com/redhat-appstudio/o11y?tab=readme-ov-file#updating-alerts) with the instructions for how to update the recording rules references.

Sure Yftach! I already created the MR for app-interface (https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/97295) and will update README as well. Thank you!

**Pull request merged**